Kernel/Storage/Partition/EBRPartitionTable.h

```diff
@@ -33,39 +33,18 @@
 
 namespace Kernel {
 
-struct [[gnu::packed]] EBRPartitionExtension
-{
-    u8 unused_area[446];
-    MBRPartitionEntry entry;
-    MBRPartitionEntry next_chained_ebr_extension;
-    MBRPartitionEntry unused[2];
-    u16 mbr_signature;
-};
-
-class EBRPartitionTable {
-
+struct EBRPartitionHeader;
+class EBRPartitionTable : public MBRPartitionTable {
 public:
-    explicit EBRPartitionTable(NonnullRefPtr<BlockDevice>);
     ~EBRPartitionTable();
 
-    bool initialize();
-    RefPtr<DiskPartition> partition(unsigned index);
+    static NonnullOwnPtr<EBRPartitionTable> try_to_initialize(const StorageDevice&);
```

Comment:
`try_to_initialize` sounds like something that can fail, yet this function returns a `NonnullOwnPtr`. So how is failure signalled to the caller? Normally, something that can fail would return an `OwnPtr` (with `nullptr` meaning failure.)

Reply:
The caller should check if the `PartitionTable` is valid by calling `table->is_valid()`. Also, if you try to get partitions from it by using the `partition` method, you will just get no value (I use the `Optional` container for such purpose) :)

Reply:
I changed the code to use `AK::Result` and `OwnPtr`. Seems fine now :)